src/core/meshcop/tcat_agent.cpp

```diff
@@ -481,6 +483,25 @@ Error TcatAgent::HandleSetActiveOperationalDataset(const Message &aIncommingMess
     return error;
 }
 
+Error TcatAgent::HandleDecomission(void)
+{
+    Error error = kErrorNone;
+
+    IgnoreReturnValue(otThreadSetEnabled(&GetInstance(), false));
+    Get<ActiveDatasetManager>().Clear();
+    Get<PendingDatasetManager>().Clear();
+
+    error = Get<Instance>().ErasePersistentInfo();
+
+#if !OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+    NetworkKey networkKey;
+    networkKey.Clear();
+    Get<KeyManager>().SetNetworkKey(networkKey);
+#endif
```

Comment:
I suggest enclosing this code block within curly braces `{ }`. An alternative solution would be to move the declaration of the `networkKey` variable to the top of the scope, before any other statements. The style guide requires that all variable declarations be placed at the beginning of the scope where they are used, and I think some toolchain code checkers may enforce this rule.


```suggestion
#if !OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
    {
        NetworkKey networkKey;
        
        networkKey.Clear();
        Get<KeyManager>().SetNetworkKey(networkKey);
    }
#endif
```

Reply:
Thanks for the review. I have applied your suggestion.